attach/cid.c

```diff
@@ -0,0 +1,235 @@
+/**
+ * @file
+ * Attachment Content-ID header functions
+ *
+ * @authors
+ * Copyright (C) 2022 David Purton <dcpurton@marshwiggle.net>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```

Comment:
We need an `@page` Doxygen comment here.
Something minimal will do.

In `main.c`, these comments link us to `attach/lib.h`
https://github.com/neomutt/neomutt/blob/af90c326a3156b509cb371f6af908fd73a72a31c/main.c#L50-L60

and in `attach/lib.h`, these comments link us to the `@page` in each source file
https://github.com/neomutt/neomutt/blob/af90c326a3156b509cb371f6af908fd73a72a31c/attach/lib.h#L28-L37